### PR TITLE
fixes neutral building destruction causing crashes when calling onStaticBuildingDestroyed if the neutral tile is offcenter

### PR DESCRIPTION
**Pull request by Cmccrave**

### CODE REVIEW

File: BWEM/src/neutral.cpp [hunk 1]
```diff
@@ -96,6 +96,9 @@ void Neutral::RemoveFromTiles()
 		auto & tile = MapImpl::Get(GetMap())->GetTile_(TopLeft() + TilePosition(dx, dy));
 		bwem_assert(tile.GetNeutral());
 
+        if (!tile.GetNeutral())
```
Comment (Cmccrave) [inline BWEM/src/neutral.cpp:99]: need to check before checking if == this

File: BWEM/src/neutral.cpp [hunk 2]
```diff
@@ -104,7 +107,7 @@ void Neutral::RemoveFromTiles()
 		else
 		{
 			Neutral * pPrevStacked = tile.GetNeutral();
-			while (pPrevStacked != nullptr && pPrevStacked->NextStacked() != this)
+			while (pPrevStacked->NextStacked() != this)
```
Comment (Cmccrave) [inline BWEM/src/neutral.cpp:110]: redundant check, don't need to check again since we check above

File: BWEM/src/neutral.cpp [hunk 3]
```diff
@@ -131,7 +134,7 @@ vector<const Area *> Neutral::BlockedAreas() const
 {
 	vector<const Area *> Result;
 	for (WalkPosition w : m_blockedAreas) {
-		auto area = GetMap()->GetArea(w);
+		auto area = GetMap()->GetNearestArea(w);
```
Comment (Cmccrave) [inline BWEM/src/neutral.cpp:137]: find nearest area for a blocking neutral that is offcenter

File: BWEM/src/neutral.cpp (hunk 1 above)
Comment (N00byEdge) [inline BWEM/src/neutral.cpp:99]: Inconsistent indentation

**Review by N00byEdge — CHANGES_REQUESTED** [review]

**Review by N00byEdge — APPROVED** [review]

**Pull request merged**